### ISSUE TITLE
Common surfaces issue when using chamferedCylinder with difference()

### ISSUE THREAD

**Issue by MarkusNaeher** [SebiTimeWaster/Chamfers-for-OpenSCAD#3]
This file demonstrates the issue:

```
$fn = 120;

include <Chamfers-for-OpenSCAD/Chamfer.scad>;

difference() {
  cube( [ 20, 20, 10 ] );
  translate( [ 10, 10, -1 ] ) chamferCylinder( r = 5, h = 12, ch = - 2, ch2 = - 2 );
  translate( [ -1, -1, -1 ] ) cube( [ 22, 11, 12 ] );
}

translate( [ 30, 10, -1 ] ) chamferCylinder( r = 5, h = 12, ch = - 2, ch2 = - 2 );
```
Although I've made the chamferedCylinder a bit higher than the cube, the preview (F5) shows a hole that does not go through.

Suggestion for a fix:

**Edit:**
I just realized that my suggested fix works only for negative chamfers. This is the one that works for both:

```
117c117
<         translate([0, 0, abs(chamferHeight)]) cylinder(height - abs(chamferHeight2) - abs(chamferHeight), r1 = radius, r2 = radius2, $fn = circleSegments(max(radius, radius2), quality));
---
>         translate([0, 0, chamferHeight > 0 ? chamferHeight : 0]) cylinder(height - (chamferHeight2 > 0 ? chamferHeight2 : 0) - (chamferHeight > 0 ? chamferHeight : 0), r1 = radius, r2 = radius2, $fn = circleSegments(max(radius, radius2), quality));
```

**Comment by SebiTimeWaster** [comment]
Hi, are you talking about the artifacts in preview mode? Because i cant see anything wrong in the rendering:
![image](https://user-images.githubusercontent.com/3726309/106053733-6dc2dc00-60eb-11eb-8be0-eab1d32d95ca.png)

**Comment by MarkusNaeher** [comment]
Yes, I'm talking about preview.  
It bothers me because usually I'm working in preview mode 95% of the time until the model is finished, and sometimes the artifacts are in the way, i.e. you don't see what's inside / behind. This forces me to use render more often in between the development process.

**Comment by SebiTimeWaster** [comment]
i mean for a negative chamfer its possible to do that, but for a positive one it would result in overlapping geometries.

**Comment by MarkusNaeher** [comment]
For positive chamfers, you could put a hull() around it.

**Comment by SebiTimeWaster** [comment]
You sent me down some rabbit hole, i can tell you. 

Here is the formal confirmation of the changes:
![image](https://user-images.githubusercontent.com/3726309/106213971-1c3f4d80-61ce-11eb-8636-9b4038a05a00.png)

I just released version "v1.1".

**Comment by RichardBronosky** [comment]
> You sent me down some rabbit hole

You are a wizard, @SebiTimeWaster. A well-named legend.
☮️❤️🌈🧘🏽🕉☸️☯️
